ion-schema/src/model/type_reference.rs

```diff
@@ -35,20 +42,18 @@ impl TypeReference {
         self.type_name.as_str()
     }
 
-    /// Returns an immutable borrow of the [TypeDefinition] to which this [TypeReference] refers.
-    pub fn get(&self) -> Option<&TypeDefinition> {
-        todo!()
+    pub(crate) fn type_coordinates(&self) -> Option<TypeCoordinates> {
+        self.resolved_type_coordinates
     }
 
-    // TODO: functions for setting the resolved type
+    pub(crate) fn resolve_type_coordinates(&mut self, type_coordinates: Option<TypeCoordinates>) {
+        self.resolved_type_coordinates = type_coordinates
+    }
 }
 
 impl<T: Into<String>> From<T> for TypeReference {
     fn from(value: T) -> Self {
-        TypeReference {
-            schema_id: None,
-            type_name: value.into(),
-        }
+        TypeReference::new(value)
```

Comment:
🪧 This change and all that follow are just switching out manual construction to using the constructor functions.